Library/Homebrew/cmd/test-bot.rb

```diff
@@ -406,7 +412,8 @@ def formula formula_name
         CompilerSelector.select_for(formula)
       rescue CompilerSelectionError => e
         unless installed_gcc
-          test "brew", "install", "gcc"
+          # Don't care about e.g. bottle failures
+          run_as_not_developer { test "brew", "install", "gcc" }
```

Comment:
I'd say the actual fix here is to only do this if the `gcc` formula has been changed?